src/chan_vese.jl

```diff
@@ -0,0 +1,258 @@
+            C₄ = 1. / sqrt(ϵ + (𝚽ᵢ₊ - 𝚽ᵢ₋)^2 / 4. + (𝚽₀ - 𝚽ⱼ₋)^2)
+
+            K = 𝚽ᵢ₊ * C₁ + 𝚽ᵢ₋ * C₂ + 𝚽ⱼ₊ * C₃ + 𝚽ⱼ₋ * C₄
+            δₕ = h / (h^2 + 𝚽₀^2) # Regularised Dirac function
+            difference_from_average = - λ₁ * (u₀ - c₁) ^ 2 + λ₂ * (u₀ - c₂) ^ 2
+
+            𝚽ⁿ⁺¹[idx] = 𝚽 = (𝚽₀ + Δt * δₕ * (μ * K + difference_from_average)) / (1. + μ * Δt * δₕ * (C₁ + C₂ + C₃ + C₄))
+            diff += (𝚽 - 𝚽₀)^2
+        end
+
+        del = sqrt(diff / area)
+
+        if reinitial_flag
+            # Reinitialize 𝚽 to be the signed distance function to its zero level set
+            reinitialize(𝚽ⁿ⁺¹, 𝚽ⁿ, Δt, h)
+        else
+            𝚽ⁿ .= 𝚽ⁿ⁺¹
+        end
+  
+        iter += 1
+    end
+
+    return 𝚽ⁿ .> 0
+end
+
+function initial_level_set(shape::Tuple)
+    x₀ = reshape(collect(0:shape[begin]-1), shape[begin], 1)
+    y₀ = reshape(collect(0:shape[begin+1]-1), 1, shape[begin+1])
+    𝚽₀ = @. sin(pi / 5 * x₀) * sin(pi / 5 * y₀)
+end
+
+function calculate_averages(img::AbstractArray{T, N}, H𝚽::AbstractArray{S, N}, area::Int64, ∫u₀::Float64) where {T<:Real, S<:Bool, N}
+    ∫u₀H𝚽 = 0
+    ∫H𝚽 = 0
+    for i in eachindex(img)
+        if H𝚽[i]
+            ∫u₀H𝚽 += img[i]
+            ∫H𝚽 += 1
+        end
+    end
+    ∫H𝚽ⁱ = area - ∫H𝚽
+    ∫u₀H𝚽ⁱ = ∫u₀ - ∫u₀H𝚽
+    c₁ = ∫u₀H𝚽 / max(1, ∫H𝚽)
+    c₂ = ∫u₀H𝚽ⁱ / max(1, ∫H𝚽ⁱ)
+
+    return c₁, c₂
+end
+
+function calculate_reinitial(𝚽::AbstractArray{T, M}, 𝚿::AbstractArray{T, M}, Δt::Float64, h::Float64) where {T<:Real, M}
+    ϵ = 1e-8
+
+    s, t = first(CartesianIndices(𝚽))[1], first(CartesianIndices(𝚽))[2]
+    m, n = last(CartesianIndices(𝚽))[1], last(CartesianIndices(𝚽))[2]
+
+    for idx in CartesianIndices(𝚽)
+        i₊ = idx[2] != n ? idx + CartesianIndex(0, 1) : idx
+        j₊ = idx[1] != m ? idx + CartesianIndex(1, 0) : idx
+        i₋ = idx[2] != t ? idx - CartesianIndex(0, 1) : idx
+        j₋ = idx[1] != s ? idx - CartesianIndex(1, 0) : idx
+        𝚽₀  = 𝚽[idx]               # 𝚽(i, j)
+        𝚽ᵢ₊ = 𝚽[i₊]                # 𝚽(i + 1, j)
+        𝚽ⱼ₊ = 𝚽[j₊]                # 𝚽(i, j + 1)
+        𝚽ᵢ₋ = 𝚽[i₋]                # 𝚽(i - 1, j)
+        𝚽ⱼ₋ = 𝚽[j₋]                # 𝚽(i, j - 1)
+
+        a = (𝚽₀ - 𝚽ᵢ₋) / h
+        b = (𝚽ᵢ₊ - 𝚽₀) / h
+        c = (𝚽₀ - 𝚽ⱼ₋) / h
+        d = (𝚽ⱼ₊ - 𝚽₀) / h
+
+        a⁺ = max(a, 0)
+        a⁻ = min(a, 0)
+        b⁺ = max(b, 0)
+        b⁻ = min(b, 0)
+        c⁺ = max(c, 0)
+        c⁻ = min(c, 0)
+        d⁺ = max(d, 0)
+        d⁻ = min(d, 0)
+
+        G = 0
+        if 𝚽₀ > 0
+            G += sqrt(max(a⁺^2, b⁻^2) + max(c⁺^2, d⁻^2)) - 1
+        elseif 𝚽₀ < 0
+            G += sqrt(max(a⁻^2, b⁺^2) + max(c⁻^2, d⁺^2)) - 1
+        end
+        sign𝚽 = 𝚽₀ / sqrt(𝚽₀^2 + ϵ)
+        𝚿[idx] = 𝚽₀ - Δt * sign𝚽 * G
+    end
+
+    return 𝚿
+end
+
+function reinitialize(𝚽::AbstractArray{T, M}, 𝚿::AbstractArray{T, M}, Δt::Float64, h::Float64, max_reiter::Int64=5) where {T<:Real, M}
```

Comment:
* 32-bit machine
* in-place function usually add `!` suffix

```suggestion
function reinitialize!(𝚽::AbstractArray{T, M}, 𝚿::AbstractArray{T, M}, Δt::Float64, h::Float64, max_reiter::Int=5) where {T<:Real, M}
```